numpy/core/arrayprint.py

```diff
@@ -1220,16 +1221,16 @@ def format_float_positional(x, precision=None, unique=True,
 
 
 class IntegerFormat:
-    def __init__(self, data):
+    def __init__(self, data, sign=None):
         if data.size > 0:
-            max_str_len = max(len(str(np.max(data))),
+            self.max_str_len = max(len(str(np.max(data))),
                               len(str(np.min(data))))
```

Comment:
This is incorrect now when the sign is forced and the value is larger than 0 you need an additional character!  (Would be nice to add a test.)